Modules/_sqlite/connection.c

```diff
@@ -249,7 +249,9 @@ pysqlite_connection_init_impl(pysqlite_Connection *self, PyObject *database,
     if (self->initialized) {
         PyTypeObject *tp = Py_TYPE(self);
         tp->tp_clear((PyObject *)self);
-        connection_close(self);
+        if (connection_close(self) < 0) {
+            return -1;
```

Comment:
Is it safe to keep `initialized` true after calling `tp_clear()`?

Reply:
I don't think so; we should set `initialized = 0` after calling `tp_clear`.